### PR TITLE
Detekt 1.18.1

### PR DESCRIPTION
**Pull request by Ozsie**
Updated to Detekt 1.18.1, plus new versions of other dependencies

### CODE REVIEW

File: .github/workflows/pre-merge.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        jdk: [8, 11]
+        jdk: [8, 11, 15, 16]
```
Comment (cortinico) [inline .github/workflows/pre-merge.yaml:16]:
```suggestion
        jdk: [8, 11, 16]
```
We generally support only LTS and the latest released version

File: pom.xml [hunk 2]
```diff
@@ -38,6 +38,11 @@
             <name>jcenter</name>
             <url>https://jcenter.bintray.com</url>
```
Comment (cortinico) [inline pom.xml:39]: Please remove this repository as well as you're adding MavenCentral

**Review by cortinico — COMMENTED** [review]
Thanks for your contribution 🙏

File: pom.xml (hunk 2 above)
Comment (Ozsie) [reply pom.xml]: Hmm, I'm not sure why I even added that one, it should work without it as well, I've committed a change to this.

File: .github/workflows/pre-merge.yaml (hunk 1 above)
Comment (Ozsie) [reply .github/workflows/pre-merge.yaml]: I've updated this as well

File: pom.xml [hunk 3]
```diff
@@ -19,41 +19,25 @@
...
 
-    <repositories>
-        <repository>
-            <id>central</id>
-            <name>jcenter</name>
-            <url>https://jcenter.bintray.com</url>
-        </repository>
-    </repositories>
-
     <pluginRepositories>
```
Comment (Ozsie) [inline pom.xml:35]: I think this part could be removed as well, but it might belong in its own PR?

**Review by schalkms — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (cortinico) [reply pom.xml:35]: Yes can you change also this to Maven Central? Now it's working as JCenter is still mirroring Central but it will break in the future.

Since you already tackled this problem for the build dependencies, makes sense to do it as well for the plugin repos.

Comment (Ozsie) [reply pom.xml:35]: I'll remove that as well then.

File: pom.xml (hunk 2 above)
Comment (cortinico) [reply pom.xml]: I'm unsure it will work without any repository defined

Comment (cortinico) [reply pom.xml]: It seems like it does actually 👀

Comment (Ozsie) [reply pom.xml]: Yeah, it should work fine if all you need is Maven Central, the Maven Super pom defines that.

**Review by cortinico — APPROVED** [review]
